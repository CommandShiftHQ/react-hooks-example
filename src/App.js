import './App.css';
import BooksContainer from './components/BooksContainer';
import Header from './components/Header';
import { useEffect, useState, createContext } from 'react';
import axios from 'axios';

export const BooksContext = createContext();

function App() {
  const [booksData, setBooksData] = useState(null);
  const [subject, setSubject] = useState('thriller');

  useEffect(() => {
    axios
      .get(`http://openlibrary.org/subjects/${subject}.json?limit=5`)
      .then(response => setBooksData(response.data));
  }, [subject]);

  function handleSetSubject(event) {
    setSubject(event.target.value);
  }

  const booksDataValue = { booksData, handleSetSubject };

  return (
    <BooksContext.Provider value={booksDataValue}>
      <div className="App">
        <Header
        // handleSetSubject={handleSetSubject}
        />
        <BooksContainer
        // booksData={booksData}
        />
      </div>
    </BooksContext.Provider>
  );
}

export default App;
