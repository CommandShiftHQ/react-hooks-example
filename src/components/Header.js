import React, { useContext } from 'react';
import { BooksContext } from '../App';

const Header = () =>
  // { handleSetSubject }
  {
    const { handleSetSubject } = useContext(BooksContext);
    return (
      <div>
        <h1>Book search</h1>
        <label>
          Select your favourite genre below
          <select onChange={handleSetSubject}>
            <option value="thriller">Thriller</option>
            <option value="fantasy">Fantasy</option>
            <option value="cats">Cats</option>
            <option value="science">Science</option>
          </select>
        </label>
      </div>
    );
  };

export default Header;
