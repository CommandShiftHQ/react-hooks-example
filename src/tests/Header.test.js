import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../components/Header';
import { BooksContext } from '../App';
import renderer from 'react-test-renderer';

const booksDataValue = {};

describe('Header', () => {
  it('contains title', () => {
    render(
      <BooksContext.Provider value={booksDataValue}>
        <Header />
      </BooksContext.Provider>,
    );
    expect(screen.getByText('Book search')).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BooksContext.Provider value={booksDataValue}>
          <Header />
        </BooksContext.Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
