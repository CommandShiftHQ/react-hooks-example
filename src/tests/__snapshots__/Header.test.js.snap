// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <h1>
    Book search
  </h1>
  <label>
    Select your favourite genre below
    <select>
      <option
        value="thriller"
      >
        Thriller
      </option>
      <option
        value="fantasy"
      >
        Fantasy
      </option>
      <option
        value="cats"
      >
        Cats
      </option>
      <option
        value="science"
      >
        Science
      </option>
    </select>
  </label>
</div>
`;
